receiver/prometheusexecreceiver/receiver_test.go

```diff
@@ -113,3 +120,663 @@ func assertTwoUniqueValuesScraped(t *testing.T, metricsSlice []pdata.Metrics) {
+		{
+			name: "normal config",
+			config: &Config{
+				ScrapeInterval: 90 * time.Second,
+				Port:           9104,
+				SubprocessConfig: subprocessmanager.SubprocessConfig{
+					Command: "mysqld_exporter",
+					Env: []subprocessmanager.EnvConfig{
+						{
+							Name:  "DATA_SOURCE_NAME",
+							Value: "password:username@(url:port)/dbname",
+						},
+					},
+				},
+			},
+			wantReceiverConfig: &prometheusreceiver.Config{
+				PrometheusConfig: &config.Config{
+					ScrapeConfigs: []*config.ScrapeConfig{
+						{
+							ScrapeInterval:  model.Duration(90 * time.Second),
+							ScrapeTimeout:   model.Duration(10 * time.Second),
+							Scheme:          "http",
+							MetricsPath:     "/metrics",
+							JobName:         "mysqld",
+							HonorLabels:     false,
+							HonorTimestamps: true,
+							ServiceDiscoveryConfig: sdconfig.ServiceDiscoveryConfig{
+								StaticConfigs: []*targetgroup.Group{
+									{
+										Targets: []model.LabelSet{
+											{model.AddressLabel: model.LabelValue("localhost:9104")},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantSubprocessConfig: &subprocessmanager.SubprocessConfig{
+				Command: "mysqld_exporter",
+				Env: []subprocessmanager.EnvConfig{
+					{
+						Name:  "DATA_SOURCE_NAME",
+						Value: "password:username@(url:port)/dbname",
+					},
+				},
+			},
+		},
+		{
+			name: "lots of defaults",
+			config: &Config{
+				ScrapeInterval: 60 * time.Second,
+				SubprocessConfig: subprocessmanager.SubprocessConfig{
+					Command: "postgres_exporter",
+					Env: []subprocessmanager.EnvConfig{
+						{
+							Name:  "DATA_SOURCE_NAME",
+							Value: "password:username@(url:port)/dbname",
+						},
+					},
+				},
+			},
+			wantReceiverConfig: &prometheusreceiver.Config{
+				PrometheusConfig: &config.Config{
+					ScrapeConfigs: []*config.ScrapeConfig{
+						{
+							ScrapeInterval:  model.Duration(60 * time.Second),
+							ScrapeTimeout:   model.Duration(10 * time.Second),
+							Scheme:          "http",
+							MetricsPath:     "/metrics",
+							JobName:         "postgres",
+							HonorLabels:     false,
+							HonorTimestamps: true,
+							ServiceDiscoveryConfig: sdconfig.ServiceDiscoveryConfig{
+								StaticConfigs: []*targetgroup.Group{
+									{
+										Targets: []model.LabelSet{
+											{model.AddressLabel: model.LabelValue("localhost:0")},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantSubprocessConfig: &subprocessmanager.SubprocessConfig{
+				Command: "postgres_exporter",
+				Env: []subprocessmanager.EnvConfig{
+					{
+						Name:  "DATA_SOURCE_NAME",
+						Value: "password:username@(url:port)/dbname",
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range configTests {
+		t.Run(test.name, func(t *testing.T) {
+			got := getSubprocessConfig(test.config)
+			assert.Equal(t, test.wantSubprocessConfig, got)
+		})
+	}
+}
+
+func TestExtractName(t *testing.T) {
```

Comment:
There seems to be a lot of redundant data in these test cases which makes it unnecessarily hard to tell what's going on here. Can you remove everything that's not needed (presumably just typeval/nameval)?